src/bgw/scheduler.c

```diff
@@ -671,11 +682,12 @@ ts_bgw_scheduler_process(int32 run_for_interval_ms,
 {
 	TimestampTz start = ts_timer_get_current_timestamp();
 	TimestampTz quit_time = DT_NOEND;
-	MemoryContext scheduler_mctx = CurrentMemoryContext;
+	scratch_mctx = AllocSetContextCreate(CurrentMemoryContext, "Scratch", ALLOCSET_DEFAULT_SIZES);
```

Comment:
It might be unclear what scratch actually means here. If I am not mistaken, this names can be seen outside when debugging memory contexts. Maybe add a scheduler prefix, etc?